test/java/seedu/addressbook/logic/LogicTest.java

```diff
@@ -90,7 +90,9 @@ private void assertCommandBehavior(String inputCommand,
         //Confirm the state of data is as expected
         assertEquals(expectedAddressBook, addressBook);
         assertEquals(lastShownList, logic.getLastShownList());
-        assertEquals(addressBook, saveFile.load());
+        if (inputCommand.equals("add") || inputCommand.equals("clear") || inputCommand.equals("delete")) {
+            assertEquals(addressBook, saveFile.load());
+        }
     }
```

Comment:
Perhaps you can also add another test to check if the file was not updated for non-mutating commands, for example checking the last update time of the file.